### ISSUE TITLE
Unit support for fitting unsupported in compound models

### ISSUE THREAD

**Issue by nmearl** [astropy/astropy#7314]
Unit support does not translate to compound models even when all sub models support fitting with units.

```python
In [54]: from astropy.modeling.models import Gaussian1D
    ...: from collections import OrderedDict
    ...:
    ...: class Gaussian(Gaussian1D):
    ...:     def _parameter_units_for_data_units(self, inputs_unit, outputs_unit):
    ...:         return OrderedDict([('mean', inputs_unit['x']),
    ...:                             ('stddev', inputs_unit['x']),
    ...:                             ('amplitude', outputs_unit['y'])])
    ...:

In [55]: g = Gaussian()

In [56]: g._supports_unit_fitting
Out[56]: True

In [57]: g2 = Gaussian() + Gaussian()

In [58]: g2._supports_unit_fitting
Out[58]: False
```

**Edit**: seems `_supports_unit_fitting` is [explicitly set as `False` for compound models](https://github.com/astropy/astropy/blob/ed3edd4dae2f8d9e537a295b4d418678cec3e12d/astropy/modeling/core.py#L2852).

**Comment by pllim** [comment]
Does this supersede your older problem at #6816 ? Just curious.

**Comment by nmearl** [comment]
No.

**Comment by nmearl** [comment]
So, this seems to be a regression between astropy 3.0.1 and astropy 3.0. Running the script above in 3.0 succeeds. I think @astrofrog was working on the model fitting stuff?

**Comment by Cadair** [comment]
We explicitly turned it off, because it dosen't work yet: https://github.com/astropy/astropy/pull/6952/commits/9fb8cec6a7d51e91d0ad6876e33a848067942cd6

**Comment by nden** [comment]
Fixed in #8369
